### PR TITLE
[SPARK-34542][BUILD] Upgrade Parquet to 1.12.0

### PR DESCRIPTION
**Pull request by wangyum**
### What changes were proposed in this pull request?

Parquet 1.12.0 New Feature
- PARQUET-41 - Add bloom filters to parquet statistics
- PARQUET-1373 - Encryption key management tools
- PARQUET-1396 - Example of using EncryptionPropertiesFactory and DecryptionPropertiesFactory
- PARQUET-1622 - Add BYTE_STREAM_SPLIT encoding
- PARQUET-1784 - Column-wise configuration
- PARQUET-1817 - Crypto Properties Factory
- PARQUET-1854 - Properties-Driven Interface to Parquet Encryption

Parquet 1.12.0 release notes:
https://github.com/apache/parquet-mr/blob/apache-parquet-1.12.0/CHANGES.md

### Why are the changes needed?

- Bloom filters to improve filter performance
- ZSTD enhancement

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing unit test.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Is the UT failure meaningful? Or, do you want to retry simply, @wangyum ?

File: pom.xml [hunk 1]
```diff
@@ -291,6 +291,17 @@
         <enabled>false</enabled>
       </snapshots>
     </repository>
+    <repository>
```
Comment (srowen) [inline pom.xml:294]: This would be removed before committing right - this is just to test 1.12 before release?

Comment (wangyum) [reply pom.xml:294]: Yes.

**Comment by maropu** [comment]
Note: I ran TPCDS queries (sf=20) based on this PR and there was no valid performance regression.

**Comment by dongjoon-hyun** [comment]
Thank you, @maropu !
> Note: I ran TPCDS queries (sf=20) based on this PR and there was no valid performance regression.

**Comment by dongjoon-hyun** [comment]
Hi, @wangyum . Could you update this PR with Apache Parquet 1.12.0 RC3?

**Comment by wangyum** [comment]
> Could you update this PR with Apache Parquet 1.12.0 RC3?

Done

**Comment by dongjoon-hyun** [comment]
Thank you so much!

**Comment by dongjoon-hyun** [comment]
Retest this please

**Comment by wangyum** [comment]
retest this please.

**Comment by dongjoon-hyun** [comment]
BTW, after SBT dependency override, the one failure at `SparkSubmitUtilsSuite` looks irrelevant. Thanks, @wangyum .
cc @iemejia

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @wangyum .
Parquet 1.12.0 is finally released. Could you use the official one by removing the pom file repository change?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @wangyum , @maropu , @srowen .
- @maropu checked that there is no performance regression in TPCDS cases (https://github.com/apache/spark/pull/31649#issuecomment-787914596).
- I hope we can test this in `master` branch actively for Apache Spark 3.2.0 timeframe (July)
- We can revert this easily if there is an issue. Also, we can give feedbacks to `Apache Parquet` community to make a maintenance release, 1.12.1.

cc @dbtsai , @holdenk , @viirya , @sunchao , @ggershinsky, @attilapiros , @gatorsmile , @cloud-fan

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.2.0 on July 2021.

**Pull request closed without merging**